### PR TITLE
cmd click on recent notebook #3840

### PR DESCRIPTION
**Pull request by kpaharelau**
Use window.open for recent items from control page

### CODE REVIEW

File: core/src/main/web/app/helpers/helper.js [hunk 1]
```diff
@@ -205,6 +205,28 @@
       getBaseUrl: function() {
         return bkUtils.getBaseUrl();
       },
+      openNotebookInNewWindow: function (notebookUri, uriType, readOnly, format) {
```
Comment (scottdraves) [inline core/src/main/web/app/helpers/helper.js]: this can't be right we need a clean way.

**Comment by scottdraves** [comment]
code is clean now, but the new window functionality should only apply if the cmd key is down (ctrl on pc), just like for normal links.

File: core/src/main/web/app/helpers/helper.js [hunk 2]
```diff
@@ -205,6 +205,22 @@
...
+
+        var path = "/open?uri=" + window.encodeURIComponent(notebookUri);
+        if (uriType) {
+          path += "&type=" + uriType;
```
Comment (tomca32) [inline core/src/main/web/app/helpers/helper.js]: No need to do this manually, you can use $httpParamSerializer.

https://docs.angularjs.org/api/ng/service/$httpParamSerializer

Docs are not great, here's an SO example:

http://stackoverflow.com/a/34039330/2341467

File: core/src/main/web/app/helpers/helper.js [hunk 3]
```diff
@@ -205,6 +205,22 @@
       getBaseUrl: function() {
         return bkUtils.getBaseUrl();
       },
+      openNotebookInNewWindow: function (notebookUri, uriType, readOnly, format) {
+
+        var path = "/open?uri=" + window.encodeURIComponent(notebookUri);
```
Comment (tomca32) [inline core/src/main/web/app/helpers/helper.js]: Since $httpParamSerializer will do the encoding, you can use it for uri param also.

**Comment by scottdraves** [comment]
:+1:

**Comment by scottdraves** [comment]
kick jenkins

**Pull request merged**